source/opt/mem_pass.cpp

```diff
@@ -212,11 +216,14 @@ void MemPass::DCEInst(ir::Instruction* inst) {
   }
 }
 
-void MemPass::ReplaceAndDeleteLoad(ir::Instruction* loadInst, uint32_t replId) {
+void MemPass::ReplaceAndDeleteLoad(ir::Instruction* loadInst,
+                                   uint32_t replId,
+                                   const function<void(
+                                       ir::Instruction *)>& call_back) {
```

Comment:
Likewise here.  Default to an empty functor for the callback?

Reply:
I have deleted this function because every place that uses it needed it functionality split up.